x-pack/plugins/security_solution/public/one_discover/app_wrapper/index.tsx

```diff
@@ -0,0 +1,125 @@
+ * 2.0.
+ */
+import React, { useMemo } from 'react';
+import { ExpandableFlyoutProvider } from '@kbn/expandable-flyout';
+import { Provider as ReduxStoreProvider } from 'react-redux';
+import { EuiThemeProvider } from '@kbn/kibana-react-plugin/common';
+import { KibanaContextProvider, useKibana } from '@kbn/kibana-react-plugin/public';
+import type { DiscoverServices } from '@kbn/discover-plugin/public/build_services';
+import { NavigationProvider } from '@kbn/security-solution-navigation';
+import type { CoreStart } from '@kbn/core/public';
+import type { SecuritySolutionAppWrapperFeature } from '@kbn/discover-shared-plugin/public';
+import { APP_ID } from '../../../common';
+import { SecuritySolutionFlyout } from '../../flyout';
+import { StatefulEventContext } from '../../common/components/events_viewer/stateful_event_context';
+import type { SecurityAppStore } from '../../common/store';
+import { ReactQueryClientProvider } from '../../common/containers/query_client/query_client_provider';
+import type { StartPluginsDependencies, StartServices } from '../../types';
+import { MlCapabilitiesProvider } from '../../common/components/ml/permissions/ml_capabilities_provider';
+import { UserPrivilegesProvider } from '../../common/components/user_privileges/user_privileges_context';
+import { DiscoverInTimelineContextProvider } from '../../common/components/discover_in_timeline/provider';
+import { UpsellingProvider } from '../../common/components/upselling_provider';
+import { ConsoleManager } from '../../management/components/console';
+import { AssistantProvider } from '../../assistant/provider';
+import { ONE_DISCOVER_SCOPE_ID } from '../constants';
+
+export const createSecuritySolutionDiscoverAppWrapperGetter = ({
+  core,
+  services,
+  plugins,
+}: {
+  core: CoreStart;
+  services: StartServices;
+  plugins: StartPluginsDependencies;
+}) => {
+  const getSecuritySolutionDiscoverAppWrapper: Awaited<
+    ReturnType<SecuritySolutionAppWrapperFeature['getWrapper']>
+  > = ({ store }) => {
+    return function SecuritySolutionDiscoverAppWrapper({ children }) {
+      const discoverServices = useKibana().services as DiscoverServices;
+
+      const CasesContext = useMemo(() => plugins.cases.ui.getCasesContext(), []);
+
+      const userCasesPermissions = useMemo(() => plugins.cases.helpers.canUseCases([APP_ID]), []);
+
+      const securitySolutionServices = useMemo(
+        () => ({
+          cases: plugins.cases,
+          telemetry: services.telemetry,
+          apm: services.apm,
+        }),
+        []
+      );
+
+      /**
+       *
+       * Uses mainly discover services and adds security solution services
+       * only when it is needed.
+       *
+       */
+      const newServicesForSecurityInDiscover = useMemo(() => {
+        return {
+          ...discoverServices,
+          ...securitySolutionServices,
+        };
+      }, [discoverServices, securitySolutionServices]);
```

Comment:
Merging services is something `KibanaContextProvider` already does internally, we don't actually need to do it manually, just doing `<KibanaContextProvider services={securitySolutionServices}>` should merge "discover" services with "securitySolution" services.

Reply:
Thanks @semd for your input. I have corrected this here: 89b4dbb7fd9